### DISCUSSION TITLE
Maintaining Quarto AUR Pre-release Packages

### DISCUSSION THREAD

**Discussion by DhruvaSambrani** [quarto-dev/quarto-cli#5423] [Feature Requests]
The pre-release bin version of quarto (from the deb file) has an AUR entry [here](https://aur.archlinux.org/packages/quarto-cli-bin-pre-release), of which I am the maintainer. 

Currently, the pre-release version is released about once a day, which is too often for manual updates. I have written a script that runs every three hours, checks if new pre-releases have released using github's api, and if it has, then updates the PKGBUILD with the correct version. While this works absolutely fine, I was just wondering if you can add this as part of the pre-release process itself to update the AUR Package.

**Comment by DhruvaSambrani** [comment]
That said, I don't mind at all continuing packaging it like now!

**Reply by cderv** [reply]
> I have written a script that runs every three hours, checks if new pre-releases have released using github's api, and if it has, then updates the PKGBUILD with the correct version.

Is this already automated on a github action or other CI service ? It seems a good process already. 
I am wondering what would the advantage of doing it on our side. Simplicity ? 

Do you have in mind a webhook signal or repository dispatch event ?

**Reply by DhruvaSambrani** [reply]
This is not automated as a github action, but rather as a script that runs on my local device periodically via a cron job.

The advantage of doing it on your side is simply not depending on an external factor (me), and nothing else.

As for triggering the script, it would just be to call it after/during the release step, however that is implemented currently. You can find following script in the AUR package too - 
```bash 
set -eu
SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
cd $SCRIPT_DIR

version=$(curl -sL \
	-H "Accept: application/vnd.github+json" \
	https://api.github.com/repos/quarto-dev/quarto-cli/releases |\
	jq "map(select(.prerelease))[0].name" -r |\
	tr -d "v")
echo $version
sha256sum=$(curl -sL https://github.com/quarto-dev/quarto-cli/releases/download/v$version/quarto-$version-checksums.txt | grep "amd.*deb" | cut -f1 -d" ")
echo $sha256sum

sed "s/:version:/$version/;s/:shasum:/$sha256sum/" PKGBUILD.template > PKGBUILD

makepkg --printsrcinfo > .SRCINFO
git add -A
git commit -am "Autorelease version $version"
git push
```

**Reply by cderv** [reply]
Thanks a lot, I see now. That is clearer. 

we have other publishing to repo that are maintained by the community, so it is not a problem for us to rely on that for a AUR 

It seems there is also a release package in AUR. do you know the maintainer ? 
It would make sense to do both release if we start doing automatically from our release process. 

AUR publishing is as simple as updating the manifest and committing, did I got it right ?

**Reply by DhruvaSambrani** [reply]
I do not know the maintainer of the release package.
AUR publishing needs 4 steps:
1. Update version
2. Update checksums
3. Update .SRCINFO
4. Push to AUR

However, this particular repo also contains the above script which does the updating from github data and pushes, so all that is necessary is to run the script.

Again, I don't mind continuing to maintain it. The above script is anyway part of the repo itself, so if I ever orphan the repo, then another person could take control and keep it running without much effort

**Reply by cderv** [reply]
Thanks a lot for taking the time to explain. 

For now, we are very glad you are maintaining this. Let's rediscuss this in the future - I'll reach out when I will reconsider our bundles release workflow. Thanks for the proposal !
